pkg/cloudprovider/aws/ami.go

```diff
@@ -44,17 +45,22 @@ func NewAMIProvider(ssm ssmiface.SSMAPI, clientSet *kubernetes.Clientset) *AMIPr
 	}
 }
 
-func (p *AMIProvider) Get(ctx context.Context, constraints *Constraints) (string, error) {
+func (p *AMIProvider) Get(ctx context.Context, constraints *Constraints, instanceTypes []cloudprovider.InstanceType) (string, error) {
 	version, err := p.kubeServerVersion(ctx)
 	if err != nil {
 		return "", fmt.Errorf("kube server version, %w", err)
 	}
-	var architectureSuffix string
+	var amiNameSuffix string
 	if *constraints.Architecture == v1alpha3.ArchitectureArm64 {
-		architectureSuffix = "-arm64"
+		amiNameSuffix = "-arm64"
 	}
-	// TODO: support for the "amazon-linux-2-gpu" AMI for nvidia use cases
-	name := fmt.Sprintf("/aws/service/eks/optimized-ami/%s/amazon-linux-2%s/recommended/image_id", version, architectureSuffix)
+	if AnyNvidiaGPUs(instanceTypes) {
```

Comment:
should it also check for AWS Neurons? 

It looks like the gpu ami ships with [AWS Neuron Container Runtime](https://docs.aws.amazon.com/eks/latest/userguide/eks-optimized-ami.html#gpu-ami)

Reply:
@bwagner5 I don't *think* so - the docs make it sound like this is (despite the name) only needed for nvidia gpus. But I would love a second opinion :-)

Reply:
It should also check for Inferentia instances. The EKS accelerated AMI works for Inf1 and Nvidia GPU instance types. It selects the correct container runtime by looking at the instance type through IMDS on startup

Reply:
Does it hard code instance types? Will this break for new instances or is there something smarter?

Reply:
@ellistarn this code doesn't hard-code instance types

Reply:
> should it also check for AWS Neurons?
> 
> It looks like the gpu ami ships with [AWS Neuron Container Runtime](https://docs.aws.amazon.com/eks/latest/userguide/eks-optimized-ami.html#gpu-ami)

done